nupic/research/temporal_memory.py

```diff
@@ -512,15 +597,17 @@ def adaptSegment(self, segment, activeSynapses, connections):
     @param segment        (int)         Segment index
     @param activeSynapses (set)         Indices of active synapses
     @param connections    (Connections) Connectivity of layer
+    @param permanenceIncrement (float)  Amount to increment active synapses
+    @param permanenceDecrement (float)  Amount to decrement inactive synapses
     """
```

Comment:
@rcrowder Perhaps. It's tricky to have generic callback functions because the logic for each is quite different and could appear at different points in the code. I agree experimental changes such as you suggest are best left for separate research repository. Once proved to be working we can consider merging into NuPIC. (That's exactly what we did for this change for example.)